File: portal/lib/http-status.ts
/**
 * HTTP status codes.
 *
 * The only file in `portal/` outside `styles/` allowed to carry bare numeric
 * literals (see scripts/lint/no_magic_numbers.py), for the same reason as
 * services/common/http_status.py: a status code is not a threshold, and
 * confining them to one module keeps them from reading like one.
 */
export const OK = 200;
export const NO_CONTENT = 204;
export const BAD_REQUEST = 400;
export const UNAUTHORIZED = 401;
export const FORBIDDEN = 403;
export const NOT_FOUND = 404;
export const UNPROCESSABLE_ENTITY = 422;
export const FAILED_DEPENDENCY = 424;
export const TOO_MANY_REQUESTS = 429;
export const INTERNAL_SERVER_ERROR = 500;
export const SERVICE_UNAVAILABLE = 503;

File: services/common/http_status.py
"""HTTP status codes.

The only module in ``services/`` allowed to carry bare numeric literals
(see ``scripts/lint/no_magic_numbers.py``). Everything else imports from here so
status codes never look like thresholds.
"""

from __future__ import annotations

from http import HTTPStatus

OK = int(HTTPStatus.OK)
CREATED = int(HTTPStatus.CREATED)
ACCEPTED = int(HTTPStatus.ACCEPTED)
NO_CONTENT = int(HTTPStatus.NO_CONTENT)
BAD_REQUEST = int(HTTPStatus.BAD_REQUEST)
UNAUTHORIZED = int(HTTPStatus.UNAUTHORIZED)
FORBIDDEN = int(HTTPStatus.FORBIDDEN)
NOT_FOUND = int(HTTPStatus.NOT_FOUND)
CONFLICT = int(HTTPStatus.CONFLICT)
UNPROCESSABLE_ENTITY = int(HTTPStatus.UNPROCESSABLE_ENTITY)
FAILED_DEPENDENCY = int(HTTPStatus.FAILED_DEPENDENCY)
TOO_MANY_REQUESTS = int(HTTPStatus.TOO_MANY_REQUESTS)
INTERNAL_SERVER_ERROR = int(HTTPStatus.INTERNAL_SERVER_ERROR)
SERVICE_UNAVAILABLE = int(HTTPStatus.SERVICE_UNAVAILABLE)
